.gitignore

```diff
@@ -1 +1,2 @@
 bazel-*
+MODULE.bazel.lock
```

Comment:
@keith raising this here as its come up in other repo - wondering about checking in lock files

@mmorel-35 was saying that its discouraged due to platform differences - im trying to understand how reproducibility is ensured without doing so (and i guess similar with the added security that using hashes provides)

Reply:
right now the lock files just aren't very stable, and across platforms i think they can differ. i think soon it will be reasonable to check them in. but bazel is otherwise using a similar algorithm to go so the dep resolution is reproducible, with the only exception I know of being if upstream yanks a version it can break (but that's not the worst thing)

Reply:
>  if upstream yanks a version it can break

but without hashes surely if either the dep pull was subject to MITM or upstream republished to same version (perhaps the registry doesnt allow that and that is the protection - similar to eg pypi) then you would have no way o knowing

dont want to block, just trying to understand the state of the art wrt bazel deps

Reply:
yea MITM for sure. It would have to MITM the registry in general.  Theoretically it's not allowed to republish the same version (but technically possible)

Reply:
hmm yeah - wondering if the registry itself should be signed a la apt etc